src/cc/tests/distributed_test.cc

```diff
@@ -610,6 +610,234 @@ TEST(DistributedTest, FullNeighborsMultipleServers)
     EXPECT_EQ(output_weights, std::vector<float>({1, 2, 1, 2, 1, 2, 1, 2}));
     EXPECT_EQ(output_counts, std::vector<uint64_t>({4, 4}));
 }
+
+std::pair<ServerList, std::shared_ptr<snark::GRPCClient>> CreateMultiServerSplitNeighborsEnvironment(std::string name)
+{
+    const size_t num_servers = 2;
+    ServerList servers;
+    std::vector<std::shared_ptr<grpc::Channel>> channels;
+
+    TestGraph::MemoryGraph m1;
+    m1.m_nodes.push_back(
+        TestGraph::Node{.m_id = 0,
+                        .m_type = 0,
+                        .m_weight = 1.0f,
+                        .m_neighbors{std::vector<TestGraph::NeighborRecord>{{1, 0, 1.0f}, {2, 0, 1.0f}}}});
+    m1.m_nodes.push_back(TestGraph::Node{
+        .m_id = 1,
+        .m_type = -1,
+        .m_weight = 1.0f,
+        .m_neighbors{std::vector<TestGraph::NeighborRecord>{{3, 0, 1.0f}, {4, 0, 1.0f}, {5, 1, 1.0f}}}});
+    TestGraph::MemoryGraph m2;
+    m2.m_nodes.push_back(
+        TestGraph::Node{.m_id = 1,
+                        .m_type = 1,
+                        .m_weight = 1.0f,
+                        .m_neighbors{std::vector<TestGraph::NeighborRecord>{{6, 1, 1.5f}, {7, 1, 3.0f}}}});
+    std::vector<TestGraph::MemoryGraph> test_graphs = {m1, m2};
+
+    for (size_t server = 0; server < num_servers; ++server)
+    {
+        TempFolder path(name);
+        auto partition = TestGraph::convert(path.path, "0_0", std::move(test_graphs[server]), 3);
+        servers.emplace_back(std::make_shared<snark::GRPCServer>(
+            std::make_shared<snark::GraphEngineServiceImpl>(path.string(), std::vector<uint32_t>{0},
+                                                            snark::PartitionStorageType::memory, ""),
+            std::shared_ptr<snark::GraphSamplerServiceImpl>{}, "localhost:0", "", "", ""));
+        channels.emplace_back(servers.back()->InProcessChannel());
+    }
+
+    return std::make_pair(std::move(servers), std::make_shared<snark::GRPCClient>(std::move(channels), 1, 1));
+}
+
+TEST(DistributedTest, NeighborSampleMultipleTypesNeighborsSpreadAcrossPartitions)
```

Comment:
Why do we need new neighbor sampling tests? I thought this PR is only for features.